src/Components/SuperAdmins/Form/index.js

```diff
@@ -3,21 +3,48 @@ import React, { useState, useEffect } from 'react';
 import Button from '../../Shared/Button';
 import Modal from '../../Shared/Modal';
 import TextInput from '../../Shared/TextInput/index';
+import Spinner from '../../Shared/Spinner/spinner';
 import { useHistory, useParams } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { postSuperAdmin, putSuperAdmin } from '../../../redux/superAdmins/thunks';
+import {
+  POST_SUPER_ADMIN_SUCCESS,
+  PUT_SUPER_ADMIN_SUCCESS
+} from '../../../redux/superAdmins/constants';
 
 const Form = () => {
   const { id } = useParams();
   const history = useHistory();
+  const dispatch = useDispatch();
+  const { error, isLoading } = useSelector((state) => state.superAdmins);
   const [isEditing, setIsEditing] = useState(false);
   const [showModal, setShowModal] = useState(false);
   const [isActionModal, setIsActionModal] = useState(false);
-  const [serverError, setServerError] = useState();
   const [superAdmin, setSuperAdmin] = useState({
     name: '',
     lastName: '',
     email: '',
     password: ''
   });
+  const superAdminParam = useSelector((state) =>
+    state.superAdmins.list.find((superAdmin) => superAdmin._id === id)
+  );
+
+  useEffect(async () => {
+    try {
+      if (id && superAdminParam) {
+        setIsEditing(true);
+        setSuperAdmin({
+          name: superAdminParam.name,
+          lastName: superAdminParam.lastName,
+          email: superAdminParam.email,
+          password: superAdminParam.password
+        });
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }, [superAdminParam]);
```

Comment:
you have to add both `id` and `superAdminParam` into the array of dependencies because those are the two things you are comparing
```suggestion
  }, id, [superAdminParam]);
```